cpp/src/io/json/json_gpu.cu

```diff
@@ -255,13 +255,16 @@ __global__ void convert_data_to_columns_kernel(parse_options_view opts,
     auto const desc =
       next_field_descriptor(current, row_data_range.second, opts, input_field_index, col_map);
     auto const value_len = static_cast<size_t>(std::max(desc.value_end - desc.value_begin, 0L));
+    const bool is_quoted =
+      *(desc.value_begin - 1) == opts.quotechar && *desc.value_end == opts.quotechar;
 
     current = desc.value_end + 1;
 
     using string_index_pair = thrust::pair<const char*, size_type>;
 
     // Empty fields are not legal values
-    if (!serialized_trie_contains(opts.trie_na, {desc.value_begin, value_len})) {
+    if (!serialized_trie_contains(opts.trie_na,
+                                  {desc.value_begin - is_quoted, value_len + is_quoted * 2})) {
```

Comment:
Don't add/subtract/multiply a raw boolean. `static_cast` to a pointer difference type.

Reply:
fixed.